ui/store/actions.ts

```diff
@@ -1239,11 +1239,13 @@ export function removeSnap(
   };
 }
 
-export async function handleSnapRequest(args: {
+export async function handleSnapRequest<
+  Params extends JsonRpcParams = JsonRpcParams,
+>(args: {
   snapId: string;
   origin: string;
   handler: string;
-  request: JsonRpcRequest;
+  request: JsonRpcRequest<Params>;
```

Comment:
Since the `request` argument is polymorphic, its generic parameter(s) should be exposed in the outer function as well.

The only exception to this would be if we want to force downstream consumers to use a certain fixed generic parameter by including it in closure.

In this case, `Params` is required to correctly type [`rawSnapRequestFunction`](https://github.com/MetaMask/metamask-extension/pull/26143/files#diff-72c459ae2e3f5674fd06869a208f0600cf4027dd04a5b1638db1677949010120R98) and ['uploadFile'](https://github.com/MetaMask/metamask-extension/pull/26143/files#diff-72c459ae2e3f5674fd06869a208f0600cf4027dd04a5b1638db1677949010120R194) in the `snap-interface.tsx` file.